File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        String[] ransomArr = ransomNote.split("");
        StringBuilder sb = new StringBuilder(magazine);
        for(int i=0;i<ransomArr.length;i++) {
            int idx = sb.indexOf(ransomArr[i]);
            if(idx < 0) return false;

            sb.deleteCharAt(idx);
        }
        return true;
    }
}

File: 0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        mem = [0] * 26
        for i in range(len(magazine)):
            mem[ord(magazine[i]) - ord('a')] += 1
        
        for i in range(len(ransomNote)):
            idx = ord(ransomNote[i]) - ord('a')
            mem[idx] -= 1
            if mem[idx] < 0:
                return False

        return True